token-lending/program/tests/repay_obligation_liquidity.rs

```diff
@@ -144,3 +144,141 @@ async fn test_success() {
+        AddObligationArgs {
+            deposits: &[(&sol_test_reserve, SOL_DEPOSIT_AMOUNT_LAMPORTS)],
+            borrows: &[(&usdc_test_reserve, USDC_BORROW_AMOUNT_FRACTIONAL)],
+            ..AddObligationArgs::default()
+        },
+    );
+
+    let (mut banks_client, payer, recent_blockhash) = test.start().await;
+
+    let initial_user_liquidity_balance = get_token_balance(&mut banks_client, usdc_test_reserve.user_liquidity_pubkey).await;
+    let initial_liquidity_supply_balance = get_token_balance(&mut banks_client, usdc_test_reserve.liquidity_supply_pubkey).await;
+
+    let mut transaction = Transaction::new_with_payer(
+        &[
+            approve(
+                &spl_token::id(),
+                &usdc_test_reserve.user_liquidity_pubkey,
+                &user_transfer_authority.pubkey(),
+                &user_accounts_owner.pubkey(),
+                &[],
+                USDC_BORROW_AMOUNT_FRACTIONAL,
+            )
+            .unwrap(),
+            refresh_obligation(
+                spl_token_lending::id(),
+                test_obligation.pubkey,
+                vec![sol_test_reserve.pubkey, usdc_test_reserve.pubkey],
+            ),
+            repay_obligation_liquidity(
+                spl_token_lending::id(),
+                USDC_BORROW_AMOUNT_FRACTIONAL,
+                usdc_test_reserve.user_liquidity_pubkey,
+                usdc_test_reserve.liquidity_supply_pubkey,
+                usdc_test_reserve.pubkey,
+                test_obligation.pubkey,
+                lending_market.pubkey,
+                user_transfer_authority.pubkey(),
+            ),
+            close_obligation_account(
+                spl_token_lending::id(),
+                test_obligation.pubkey,
+                test_obligation.owner,
+                user_accounts_owner.pubkey,
+                sol_test_reserve.user_collateral_pubkey,
+                lending_market.pubkey,
+            ),
+
+        ],
+        Some(&payer.pubkey()),
+    );
+
+    transaction.sign(
+        &[&payer, &user_accounts_owner, &user_transfer_authority],
+        recent_blockhash,
+    );
+    assert!(banks_client.process_transaction(transaction).await.is_ok());
+
+    let user_liquidity_balance = get_token_balance(&mut banks_client, usdc_test_reserve.user_liquidity.pubkey).await;
+    assert_eq!(
+        user_liquidity_balance,
+        initial_user_liquidity_balance - USDC_BORROW_AMOUNT_FRACTIONAL
+    );
+
+    let liquidity_supply_balance = get_token_balance(&mut banks_client, usdc_test_reserve.liquidity_supply_pubkey).await;
+    assert_eq!(
+        liquidity_supply_balance,
+        initial_liquidity_supply_balance + USDC_BORROW_AMOUNT_FRACTIONAL
+    );
+
+    let obligation_balance = get_token_balance(&mut banks_client, test_obligation.pubkey).await;
+    assert!(obligation_balance < 1);
+
+    let obligation = test_obligation.get_state(&mut banks_client).await;
+    assert_eq!(obligation.borrows.len(), 0);
```

Comment:
You'll also want to check that the rent lamports from the obligation went to destination